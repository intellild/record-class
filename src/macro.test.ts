import plugin from "babel-plugin-macros";
import pluginTester from "babel-plugin-tester";

pluginTester({
  plugin,
  pluginName: "macro-record-class",
  babelOptions: {
    filename: __filename,
    plugins: [["@babel/plugin-syntax-decorators", { legacy: true }]],
    presets: ["@babel/preset-typescript"],
  },
  snapshot: true,
  tests: [
    `
      import record from "../macro";
      
      @record
      class SomeRecord {
        public readonly name?: string = "";
      }
    `,
  ],
});
