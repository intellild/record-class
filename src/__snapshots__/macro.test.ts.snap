// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macro-record-class ignore fields that are not public readonly: ignore fields that are not public readonly 1`] = `

import record from "./proxy.macro";

@record
export class SomeRecord {
  public field1?: string = "";
  private readonly field2 = false;
}

      ↓ ↓ ↓ ↓ ↓ ↓

export class SomeRecord {
  field1 = "";
  field2 = false;

  constructor(partial) {}
}


`;

exports[`macro-record-class simple: simple 1`] = `

import record from "./proxy.macro";

@record
export class SomeRecord {
  public readonly name?: string = "";

  public someMethod(): void {
    //
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

export class SomeRecord {
  name;

  someMethod() {
    //
  }

  constructor(partial) {
    this.name = partial.name ?? "";
  }
}


`;
